testdata/lineage/basic-multiversion.txtar

```diff
@@ -0,0 +1,337 @@
+    version: [0, 3]
+    schema: {
+        init:        string
+        optional?:   int32
+        withDefault: *"foo" | "bar" | "baz"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            init: "some string"
+            withDefault: "baz"
+        }
+        withOptional: {
+            all:  42
+            init: "some string"
+            optional: 32
+            withDefault: "baz"
+        }
+    }
+},
+{
+    version: [1, 0]
+    schema: {
+        renamed:     string
+        optional?:   int32
+        withDefault: "foo" | *"bar" | "baz"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            renamed: "some string"
+            withDefault: "foo"
+        }
+        withOptional: {
+            all:  42
+            renamed: "some string"
+            optional: 32
+            withDefault: "bar"
+        }
+    }
+},
+{
+    version: [1, 1]
+    schema: {
+        renamed:     string
+        optional?:   int32
+        withDefault: "foo" | *"bar" | "baz" | "bing"
+    }
+    examples: {
+        withoutOptional: {
+            all:  42
+            renamed: "some string"
+            withDefault: "bing"
+        }
+        withOptional: {
+            all:  42
+            renamed: "some string"
+            optional: 32
+            withDefault: "bing"
+        }
+    }
+}]
+
+lenses: [{
+    to: [0, 3]
+    from: [1, 0]
+    input: _
+    result: {
+        init: input.renamed
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+
+        withDefault: input.withDefault
+    }
+},
+{
+    to: [0, 1]
+    from: [0, 2]
+    input: _
+    result: {
+        init: input.init
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+    }
+},
+{
+    to: [0, 2]
+    from: [0, 3]
+    input: _
+    result: {
+        init: input.init
+        all:  input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+        // TODO: withDefault: input.withDefault doesn't work
+        withDefault: "foo"
+    }
+},
+{
+    to: [1, 0]
+    from: [0, 3]
+    input: _
+    result: {
+        renamed: input.init
+        all:     input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+        // TODO: withDefault: input.withDefault doesn't work
+        withDefault: "foo"
+    }
+},
+{
+    to: [1, 0]
+    from: [1, 1]
+    input: _
+    result: {
+        renamed: input.renamed
+        all:     input.all
+        if (input.optional != _|_) {
+            optional: input.optional
+        }
+        // TODO: withDefault: input.withDefault doesn't work
+        withDefault: "foo"
+    }
+},
+{
+    to: [0, 0]
+    from: [0, 1]
+    input: _
+    result: {
+        init: input.init
+        all:  input.all
+    }
+}]
+-- out/core/instance/translate/lineage/basic-multiversion-simple-0.0->0.0.json --
```

Comment:
i am delightfully surprised by how much MORE readable the whole set is than when then JSON is formatted and spread over multiple lines! i can actually visually scan this! 🎉

Reply:
though...looking at it more, i'm finding that i kind of want to see both pre and post translation in the printed output, as otherwise i have to jump back and forth. WDYT?

Reply:
Sounds good, let's try!